plugins/module_utils/network/sonic/config/poe/poe.py

```diff
@@ -0,0 +1,675 @@
+
+            deleted_list = []
+
+            for interface_d in to_delete_list:
+                for interface_h in have["interfaces"]:
+                    if interface_d["name"] == interface_h["name"]:
+                        if interface_d.keys() == {"name"}:
+                            deleted_list.append(interface_h)
+                        else:
+                            filtered_delete = deepcopy(interface_d)
+                            for setting in interface_d:
+                                if setting not in interface_h or interface_d[setting] != interface_h[setting]:
+                                    del filtered_delete[setting]
+                            if len(filtered_delete) > 1:
+                                deleted_list.append(filtered_delete)
+            if len(deleted_list) > 0:
+                commands["interfaces"] = deleted_list
+
+        if "global" in want and "global" in have:
+            deleted_global_settings = {}
+            if "auto_reset" in want["global"] and "auto_reset" in have["global"] and want["global"]["auto_reset"] == have["global"]["auto_reset"]:
+                deleted_global_settings.update({"auto_reset": have["global"]["auto_reset"]})
+            if "power_mgmt_model" in want["global"] and "power_mgmt_model" in have["global"] and \
+                    want["global"]["power_mgmt_model"] == have["global"]["power_mgmt_model"]:
+                deleted_global_settings.update({"power_mgmt_model": have["global"]["power_mgmt_model"]})
+            if "usage_threshold" in want["global"] and "usage_threshold" in have["global"] and \
+                    want["global"]["usage_threshold"] == have["global"]["usage_threshold"]:
+                deleted_global_settings.update({"usage_threshold": have["global"]["usage_threshold"]})
+            if len(deleted_global_settings) > 0:
+                commands["global"] = deleted_global_settings
+
+        requests.extend(self.make_delete_requests(commands, have))
+        if commands and len(requests) > 0:
+            commands = update_states(commands, "deleted")
+        else:
+            commands = []
+        return commands, requests
+
+    def validate_normalize_config(self, config):
+        '''validates passed in config has values for power_limit and usage_threshold are in range as well as validate against arg spec.
+        passes back config with interface names normalized'''
+        # first removing null values so validate doesn't break
+        config = remove_none(config)
+        # validate returns validated config that is rooted at the root of argspec and with added nulls for fields of nested objects that
+        # didn't have a value passed in but some fields in the object did
+        config = validate_config(self._module.argument_spec, {"config": config})["config"]
+        # not really using the none values in this module so getting thrown out. Use empty lists for clear
+        config = remove_none(config)
```

Comment:
remove_none is used twice in validate_config because validate fails if passed in config has null/none values and argspec doesn't allow a null value and it also ends up filling up unspecified settings with none. The first call is to prevent validate from failing for stupid reasons, and the second is to prevent the nulls from getting to the rest of playbook handling. Maybe to second remove could be omitted, but I would have to comb through again to check if it breaks any logic

Reply:
Thanks for explaining. This is a valid reason.

Resolved.